create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/aws/s3/AwsS3Service.java

```diff
@@ -42,41 +40,38 @@ public AwsS3Service(AmazonS3 s3Client, String bucketName) {
      */
     @JacocoGenerated
     public AwsS3Service(Environment environment) {
-        this(AmazonS3ClientBuilder.standard().withRegion(environment.readEnv(AWS_REGION_ENV)).build(),
-                environment.readEnv(BUCKET_NAME_ENV));
+        this(S3Client.create(), environment.readEnv(BUCKET_NAME_ENV));
     }
 
     /**
      * Generate a presigned download URL.
      *
      * @param key      key
-     * @param mimeType mimeType
      * @return A presigned download URL
      * @throws ApiGatewayException exception thrown if value is missing
      */
-    public String createPresignedDownloadUrl(String key, String mimeType, Date expiration) throws ApiGatewayException {
-        try {
-            GeneratePresignedUrlRequest generatePresignedUrlRequest =
-                    createGeneratePresignedUrlRequest(key, mimeType, expiration);
-
-            return s3Client.generatePresignedUrl(generatePresignedUrlRequest).toExternalForm();
+    public PresignedUri createPresignedDownloadUrl(String key, Duration duration) throws ApiGatewayException {
+        try (var s3Presigner = S3Presigner.builder().region(Region.EU_WEST_1).s3Client(s3Client).build()) {
+            var presignGetObject = s3Presigner.presignGetObject(createPresignObjectRequest(key, duration));
+            return new PresignedUri(presignGetObject.url().toExternalForm(), Date.from(presignGetObject.expiration())
+                , null);
```

Comment:
Swapped to this version of presignUrl as is includes the expiration that aws has granted the resource.